errors/container.go

```diff
@@ -0,0 +1,185 @@
+// The Error Container entity serves a purpose for keeping track of errors,
+// details and fields information. This component can be used explicitly (for
+// example when we want to sequentially fill it with details and fields), as
+// well as implicitly (all errors that are returned from handler are
+// transformed to an Error Container, and passed as GRPCStatus to a gRPC Gateway).
+package errors
+
+import (
+	"fmt"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/golang/protobuf/proto"
+
+	"github.com/infobloxopen/atlas-app-toolkit/rpc/errdetails"
+	"github.com/infobloxopen/atlas-app-toolkit/rpc/errfields"
+)
+
+// Container struct is an entity that servers a purpose of error container and
+// consist of methods to append details, field errors and setting general
+// error code/message.
+type Container struct {
+	// details field contains an array of error details.
+	details []*errdetails.TargetInfo
+
+	// fields field contains per-field error map.
+	fields *errfields.FieldInfo
+
+	// errCode, errMessage field contain the general error message.
+	errCode    codes.Code
+	errMessage string
+
+	// errSet flag indicates whether the error was set by calling one of
+	// following methods: Set, WithDetail(s), WithField(s).
+	errSet bool
+
+	// Mapper structure performs necessary mappings.
+	Mapper
+}
+
+// NewContainer function returns a new entity of error container.
+func NewContainer(code codes.Code, format string, args ...interface{}) *Container {
+	return (&Container{}).New(code, format, args...)
+}
+
+func InitContainer() *Container {
+	return (&Container{}).New(codes.Unknown, "Unknown")
+}
+
+// Error function returns "Unknown".
+func (c Container) Error() string { return "Unknown" }
```

Comment:
1) `c Container` or `c *Container`?
2) why always "Unknown"? does it make sense to return errMessage?

Reply:
to implement error interface, Error() method should have non-pointer receiver. I left unknown, because it is the only way to indicate the misuse of container (if someone wants to return it as error).